### ISSUE TITLE
Fix pypi.yaml build

### ISSUE THREAD

**Issue by ns-rse** [claritychallenge/clarity#266]
On trying to release v0.3.0 I hit two problems

## Missing `:` in `pypi.yaml`

Corrected this directly on GitHub, then deleted the tag (`git tag --delete origin v0.3.0`), which reverted the release notes to Draft format and I then published the release again.

## `ERROR    InvalidDistribution: Metadata is missing required fields: Name.`

This cropped up when trying to upload to PyPI, but there are several problems as the package is build as  `UNKNOWN--0.3.1.dev0+g6108c20.d20230331`

1) The `[project] name="pyclarity"` from `pyproject.toml` is not being used.
2) This is _not_ the `0.3.0` tag that should have been applied when creating the release for the second time.
3) The commit hash is included, when using a `v0.3.0` tag on `HEAD` the version should not be ahead and so the version should not include the git commit.

This [thread](https://github.com/pypa/setuptools/issues/3269) indicates that this is a problem with Ubuntu 22.04 (Jammy). The specific solution highlighting this is [here](https://github.com/pypa/setuptools/issues/3269#issuecomment-1254507377) and suggests...

> TLDR; On Ubuntu 22.04 (Jammy) ensure the environment variable DEB_PYTHON_INSTALL_LAYOUT=deb_system before installing a PEP621 compliant project with pip and setuptools.

I shall attempt to rectify this now (the package builds using the correct tags locally but I'm not using Ubuntu 22.04).

**Comment by ns-rse** [comment]
Closing as now fixed. Took some trial and error and in the end I discovered I also needed to change the `setuptools_scm` configuration in `pyproject.toml`.  See commit [61ba51](https://github.com/claritychallenge/clarity/commit/61ba51371499148ad3d13ea9df7d8ef56a0c441a).

The GitHub Action that is meant to attach the build artefacts to the release notes has failed, not quite sure why but this may be because I've been fixing this via the command line and pushing tags rather than creating them on GitHub so there was no release to attach the artifacts to.  I'll have to sort that out next time though, you can't re-release to PyPI with the same version number.  In the mean time I shall make the release here on GitHub and manually attach the artifacts.
